build.sbt

```diff
@@ -78,7 +81,8 @@ lazy val scalatraSettings = Seq(
     "-feature",
     "-language:higherKinds",
     "-language:implicitConversions",
-    "-language:existentials"
+    "-language:existentials",
+    "-release:" + javaTargetRelease(virtualAxes.?.value.getOrElse(Nil)),
```

Comment:
This change (and the accompanying one above for `javacOptions`) sets the appropriate compilation target, depending on whether this is a `javax` build (11) or a `jakarta` build (17).

Note: I just learned that Scala 2.12 is only able to emit Java 8 bytecode.
So, in this case, `-release` means "compile against a specific platform API version".
So my question is: what about dropping support for Scala 2.12?

Reply:
Regarding the compilation target to 17 for the `jakarta`, I leave it to your appreciation.
While Jetty 12 requires Java 17, I just realized that, for example, Tomcat 10.1.x does support Jakarta Servlet 6, yet with Java 11 only as requirement (https://tomcat.apache.org/whichversion.html).
So let me know if you prefer to stick to Java 11 for compatibility reasons.

Reply:
Oh, I see. It may be time to drop Scala 2.12 support.

Reply:
It would be great if Scalatra 3.1.0 can support Java 11 both for `javax` and `jakarta`.

Reply:
> It would be great if Scalatra 3.1.0 can support Java 11 both for javax and jakarta.

In this case, I would only need to revert https://github.com/scalatra/scalatra/pull/1610/commits/a475aae4d8e8791e14da6c4c8f6526de4decd5aa and Java 11 would be supported for both `javax` and `jakarta` (albeit the last one with a container different than Jetty 12; Tomcat 10.1.x for example, as I mentioned above).

Reply:
As I alluded to in my previous message, I reverted https://github.com/scalatra/scalatra/commit/a475aae4d8e8791e14da6c4c8f6526de4decd5aa, in order to support a Jakarta deployment with Tomcat 10.1.x on Java 11.

Speaking of Tomcat, I've just added a new module to Scalatra, which will give the ability to *test* the Jakarta build on Java 11.
What do you think of that?

Reply:
Okay, our options are:

- Drop Java 11 support
- Continue Java 11 support with Tomcat module

However, still some sort of workaround seems to be necessary to run CI with Java 11 for Scalatra itself and it will anyway bring complexity I"m concerning.

I think it's okay to drop Java 11 support in Scalatra 3.1.0. Upgrading Java version would be relatively easier than migration applications from javax to jakarta.

Reply:
> I think it's okay to drop Java 11 support in Scalatra 3.1.0. Upgrading Java version would be relatively easier than migration applications from javax to jakarta.

So, just for confirmation, dropping Java 11 support in the sense of compiling for Java 17?
Or keep compiling for Java 11 but stop actively testing (in CI) if that Scalatra still works in that environment?

The second option would be like:
> Scalatra 3.1.0 officially supports Java 17, but _may_ work with Java 11 (i.e. with an application server that accepts it, as long as it supports Java Servlet 4.0 or Jakarta Servlet 6, depending on the chosen build).

Reply:
Hey @takezoe, did you see my last message in this thread?
Please indicate which of the 2 option I should go forward with.

Reply:
Hey @takezoe, I hope everything's going fine on your side.
Did you have the opportunity to look at my last message?

Reply:
Sorry for my late response as my new year vacation awas ended and got back to work.

> So, just for confirmation, dropping Java 11 support in the sense of compiling for Java 17?

This is what I meant.

Reply:
OK, following your input, I've created a new PR with just the right commits: https://github.com/scalatra/scalatra/pull/1632.
If you agree, we can close this one and use the new one instead.

Reply:
Thank you! https://github.com/scalatra/scalatra/pull/1632 looks good. I just added one comment about MiMa to the new PR.